packages/web3-providers/src/Session/Session.ts

```diff
@@ -0,0 +1,27 @@
+import { type SessionType, type Session } from '../types/Session.js'
+
+export abstract class BaseSession implements Session {
+    constructor(
+        public type: SessionType,
+        public profileId: string,
+        public token: string,
+        public createdAt: number,
+        public expiresAt: number,
+    ) {}
+
+    serialize(): `${SessionType}:${string}` {
+        const body = JSON.stringify({
```

Comment:
In case this is part of some protocol, or will be signed, you may need `json-stable-stringify`